### PR TITLE
doc: Put all local dev env instructions in one place

### PR DESCRIPTION
**Pull request by Nanak360**
## Fixes Issue
#6115 

## Changes proposed
Combine the two local documentation cards into one
remove docker-compose dev card from docs page
add both docker-compose dev setup and local dev setup under local-dev doc page

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [x] My change requires changes to the documentation.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots

Here is a screenshot of the `docs/environments/local-development` page: 
LinkFree-Local-Development-Documentation

and here is a screenshot of `docs` page:
LinkFree-Documentation

## Note to reviewers
let me know if some additional changes are required or if any change in text is needed.

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/6227"></a>

### CODE REVIEW

File: pages/docs/environments/local-development.mdx [hunk 1]
```diff
@@ -1,7 +1,53 @@
...
+
+5. If you want to stop the containers, you can use the following command.
+
+```bash
+docker compose down
+```
+
+### Troubleshooting
+
+If you are getting an error, first try to rebuild the images with `docker compose up --build`. If that doesn't work, stop the containers with `docker compose down` and then remove the stopped linkfree container with `docker rm linkfree`. Then remove the images with `docker rmi linkfree-app`. Then rebuild the images with `docker compose up --build`.
```
Comment (CBID2) [inline pages/docs/environments/local-development.mdx:39]:
```suggestion
If you are getting an error, try to rebuild the images with `docker compose up --build`. If that doesn't work, stop the containers with `docker compose down` and remove the stopped linkfree container with `docker rm linkfree`. Then, remove the images with `docker rmi linkfree-app`. Lastly, rebuild the images with `docker compose up --build`.
```

File: pages/docs/environments/local-development.mdx [hunk 2]
```diff
@@ -1,7 +1,53 @@
 import DocsLayout from "@components/layouts/DocsLayout.js";
 import Link from "@components/Link";
 
-## Quickstart for local development
+<div id="docker-compose">
+
+## 1. Docker Compose (recommended)
+
+This might sound scary but it is actually the easiest solution for a local development environment.
+This environment will allow you to run the application locally with Docker Compose. Everything will be run in containers (for example MongoDB), so you don't need to install anything on your machine.
```
Comment (CBID2) [inline pages/docs/environments/local-development.mdx:9]:
```suggestion
The environment will allow you to run the application locally with Docker Compose. Everything will run in containers (e.g.,MongoDB), so you don't need to install anything on your machine.
```

File: pages/docs/environments/local-development.mdx [hunk 3]
```diff
@@ -1,7 +1,53 @@
...
+```
+
+3. You can now access the application at `http://localhost:3000`.
+4. If you make any changes to the dependencies, it's recommended to rebuild the images with the following command.
+
+```bash
+docker compose up --build
+```
+
+5. If you want to stop the containers, you can use the following command.
```
Comment (CBID2) [inline pages/docs/environments/local-development.mdx:31]:
```suggestion
5. If you want to stop the containers, you can use the following command:
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi @Nanak360! 😀 You're off to a great start with the descriptions! 😀  I just added a few suggestions to make the steps easier for new community members to understand. I hope you find them helpful! 😀

File: pages/docs/environments/local-development.mdx (hunk 3 above)
Comment (Nanak360) [reply pages/docs/environments/local-development.mdx]: Good eyes 👀. I have added your suggestions.

**Comment by eddiejaoude** [comment]
@amandamartin-dev I hope you don't mind me tagging you, as you raise the issue, I was hoping you could review and see if it is what you had in mind?

File: pages/docs/environments/local-development.mdx [hunk 4]
```diff
@@ -1,7 +1,53 @@
 import DocsLayout from "@components/layouts/DocsLayout.js";
 import Link from "@components/Link";
 
-## Quickstart for local development
+<div id="docker-compose">
```
Comment (eddiejaoude) [inline pages/docs/environments/local-development.mdx:4]: I think you can use the headings for deep linking (`##`)

File: components/layouts/DocsLayout.js [hunk 5]
```diff
@@ -41,6 +41,9 @@ export default function DocsLayout({ children, title }) {
           <div className="flex flex-grow flex-row">
             <MDXProvider components={ComponentStyle}>
               <div className="w-full max-w-7xl px-4 sm:px-6 lg:px-8 dark:text-white">
+                <h3 className="mb-8 font-bold text-xl md:text-3xl">
+                  You can setup LinkFree code base in the following two ways.
+                </h3>
```
Comment (eddiejaoude) [inline components/layouts/DocsLayout.js:46]: Does this need to be on every page?

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Thank you for the contribution 👍 I added some inline comments

File: components/layouts/DocsLayout.js (hunk 5 above)
Comment (Nanak360) [reply components/layouts/DocsLayout.js:46]: no actually we do not need it for all. I'll change and push it

File: pages/docs/environments/local-development.mdx (hunk 4 above)
Comment (Nanak360) [reply pages/docs/environments/local-development.mdx:4]: `<div id="docker-compose">` this I'm using for `https://linkfree.io/docs/environments/local-development#docker-compose` in the readme file. Having an id for this section will allow us to auto scroll to the docker-compose section when user clicks on the link.

Comment (eddiejaoude) [reply pages/docs/environments/local-development.mdx:7]: I think if we update the mdx component for the headings, then we won't need the extra html and we can use this in other places also https://github.com/EddieHubCommunity/LinkFree/tree/main/components/mdx

Comment (Nanak360) [reply pages/docs/environments/local-development.mdx]: I have added a new component named Heading and used it. Have to say it is a far better approach than what I was using.

Comment (Nanak360) [reply pages/docs/environments/local-development.mdx]: This is how it looks now: 


image


image

**Comment by amandamartin-dev** [comment]
> @amandamartin-dev I hope you don't mind me tagging you, as you raise the issue, I was hoping you could review and see if it is what you had in mind?

Apologies! Missed this notification - reviewing now.

File: pages/docs/environments/local-development.mdx (hunk 4 above)
Comment (amandamartin-dev) [reply pages/docs/environments/local-development.mdx]: Hello! There is no need to create a header component for linking from external URL or within documents as markdown supports this natively.

Requesting removal of the header component and staying with pure markdown to remain consistent throughout the project and decrease complexity by adding more code.

Here is an example of how to adjust your headings in pure markdown to allow for linking.  Note that if you have a 2 word header, the url should utilize the first word in the header for the link

[markdown with linking example](https://github.com/fefong/markdown_readme/blob/master/README.md?plain=1)

example header with url for 2 word scenario:  

LOCAL DEVELOPMENT

would be url.com/page-name#local

If you would like assistance making these changes, let me know and I"m happy to help further

**Review by amandamartin-dev — CHANGES_REQUESTED** [review]

File: components/mdx/Heading.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
+const Heading = ({ children, id, fontSize = "medium", customClasses }) => {
```
Comment (amandamartin-dev) [inline components/mdx/Heading.js:1]: Added more details in the conversation with Eddie, but this component isn't needed as it's duplicating native markdown functionality.

File: pages/docs/environments/local-development.mdx (hunk 4 above)
Comment (Nanak360) [reply pages/docs/environments/local-development.mdx]: Hey Amanda and Eddie, I have gone through all the reviews and I'm making appropriate changes.
I will be pushing my changes in some time.

File: components/mdx/Heading.js (hunk 6 above)
Comment (Nanak360) [reply components/mdx/Heading.js]: I have removed this component

**Comment by amandamartin-dev** [comment]
Hey @Nanak360 ! Saw your comment, it worked when I tested in gitpod, but was very brief. I will check again when I am back at a computer and add some code if adjustments are needed. Thank you for testing it! If I am wrong when I look again we can revert to your method. Thank you!

**Comment by Nanak360** [comment]
> You have 2 section headings that are both named Local development. One of these should be changed.

Yes, this is an issue that I have realized as well.
The new id or anchor-tag logic that I introduced will convert the text after a `##` into a kebab case.
As IDs have to be unique throughout the website, I refrained from using only the first word after `##`.
And so "Local Development" becomes "local-development".
Maybe I can change the page link to `/docs/environments/development` or `/docs/environments/code-development`.
Open to suggestions here.

> The 2 sections should be h2 or double hash (The bold text is in an h2 right now and should not be. Semantically, this is not a header of any kind)

I have taken care of this part. Both the sections are now a `##` or h2.


> Let me know if this helps get you further. I didn't want to push any code into your branch in case you were working, but am happy to do so later if you prefer.

Please do. No problemo at all! 🤝

**Comment by amandamartin-dev** [comment]
> Please do. No problemo at all! 🤝

I have pushed some changes to the docs page. Pull them in and take a look and let me know what you think.

**Comment by Nanak360** [comment]
> 
> I have pushed some changes to the docs page. Pull them in and take a look and let me know what you think.

Yeah looks good to me. The local-development link was wrong in the readme file. My mistake.
changes it and pushed it.
I think we are good with all the changes now.

File: components/mdx/ComponentStyle.js [hunk 7]
```diff
@@ -3,11 +3,13 @@ import Li  from "@components/mdx/Li";
 import Link from "@components/Link";
 import Strong from "@components/mdx/Strong";
 import BlockQuote from "@components/mdx/BlockQuote";
+import H2 from "@components/mdx/H2";
```
[the remark this answers is not in the document]
Comment (Nanak360) [reply components/mdx/ComponentStyle.js:6]: This is what's attaching an id attribute to the `##` tags.

[the remark this answers is not in the document]
Comment (amandamartin-dev) [reply components/mdx/ComponentStyle.js:6]: I don't think this is necessary but let me test with/without it first.  brb!

[the remark this answers is not in the document]
Comment (Nanak360) [reply components/mdx/ComponentStyle.js:6]: Okey dokey

[the remark this answers is not in the document]
Comment (amandamartin-dev) [reply components/mdx/ComponentStyle.js:6]: Okay, so I'm definitely going insane bc I know mdx supports this natively but it def is not working in the project for some reason.  So for now, we can def keep the component perhaps there is some reason it doesn't work here.

**Review by amandamartin-dev — APPROVED** [review]
lgtm! Thank you for the discussion and learnings

**Comment by eddiejaoude** [comment]
Great collaboration! I am going to have a play now

**Review by eddiejaoude — APPROVED** [review]
Great collaboration 💪

**Pull request merged**